common/src/native.rs

```diff
@@ -109,26 +109,24 @@ pub fn program_exists(id: ProgramId) -> bool {
     crate::program_exists(H256::from_slice(id.as_slice()))
 }
 
-pub fn insert_waiting_message(prog_id: ProgramId, msg_id: MessageId, message: CoreMessage) {
+pub fn insert_waiting_message(
+    prog_id: ProgramId,
+    msg_id: MessageId,
+    message: CoreMessage,
+    bn: u32,
```

Comment:
I would suggest to make generic over BlockNumber similar how it is defined in `frame_system::Config`. Not insisting though

Reply:
Used to have it that way before. Then got rid of the generic part for the sake of simplicity. Although personally I agree the generic way is more appropriate